src/server.cpp

```diff
@@ -3537,6 +3572,85 @@ void Server::deleteParticleSpawner(const std::string &playername, u32 id)
 	SendDeleteParticleSpawner(peer_id, id);
 }
 
+u32 Server::add3DLine(const LineParams &p)
+{
+	if (!m_env)
+		return -1;
```

Comment:
It would be more consistent to reserve the ID 0 (like in ActiveObject) so that checking for its validity is similar to pointers: `if (line_id) { .. }`